File: problems/3120-count-the-number-of-special-characters-i/solution.py
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        lower = set()
        upper = set()
        for char in word:
            if 'a' <= char <= 'z':
                lower.add(char)
            else:
                upper.add(char.lower())
        return len(lower.intersection(upper))

File: problems/3120-count-the-number-of-special-characters-i/solution.cpp
class Solution {
public:
    int numberOfSpecialChars(string word) {
        int count = 0;
        bool lowercase[26] = {false};
        bool uppercase[26] = {false};

        for (char c : word) {
            if (islower(c)) {
                lowercase[c - 'a'] = true;
            } else {
                uppercase[c - 'A'] = true;
            }
        }

        for (int i = 0; i < 26; ++i) {
            if (lowercase[i] && uppercase[i]) {
                count++;
            }
        }

        return count;
    }
};